components/threading/global_threads_link/global_threads_link.tsx

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useEffect} from 'react';
+import {Link, useRouteMatch, useLocation, matchPath} from 'react-router-dom';
+import classNames from 'classnames';
+import {useIntl} from 'react-intl';
+import {useSelector, useDispatch} from 'react-redux';
+
+import {getThreadCountsInCurrentTeam} from 'mattermost-redux/selectors/entities/threads';
+import {isCollapsedThreadsEnabled} from 'mattermost-redux/selectors/entities/preferences';
+import {getThreads} from 'mattermost-redux/actions/threads';
+
+import {t} from 'utils/i18n';
+
+import {isUnreadFilterEnabled} from 'selectors/views/channel_sidebar';
+import {useThreadRouting} from '../hooks';
+
+import ChannelMentionBadge from 'components/sidebar/sidebar_channel/channel_mention_badge';
+
+import ThreadsIcon from './threads_icon';
+
+import './global_threads_link.scss';
+
+const GlobalThreadsLink = () => {
+    const {formatMessage} = useIntl();
+    const dispatch = useDispatch();
+    const isFeatureEnabled = useSelector(isCollapsedThreadsEnabled);
+
+    const {url} = useRouteMatch();
+    const {pathname} = useLocation();
+    const threadsMatch = matchPath<{team: string; threadIdentifier?: string}>(pathname, '/:team/threads');
+    const {currentTeamId, currentUserId} = useThreadRouting();
+
+    const counts = useSelector(getThreadCountsInCurrentTeam);
+    const unreadsOnly = useSelector(isUnreadFilterEnabled);
+    const someUnreadThreads = counts?.total_unread_threads;
+
+    useEffect(() => {
+        dispatch(getThreads(currentUserId, currentTeamId, {perPage: 5}));
+    }, [currentTeamId]);
+
+    if (!isFeatureEnabled || (unreadsOnly && !threadsMatch && !someUnreadThreads)) {
```

Comment:
Looks like the feature gate doesn't stop the `dispatch` above? Should it?

Reply:
Good catch! Yes, it should